File: codeforces/0900/B_Array_Cloning_Technique.cpp
#include <bits/stdc++.h>

using i64 = long long;

void solve() {
    int n;
    std::cin >> n;

    int highest_count = 0;

    std::map<int, int> num_count;

    for (int i = 0; i < n; i++) {
        int num;
        std::cin >> num;
        num_count[num]++;
        if (num_count[num] > highest_count) {
            highest_count = num_count[num];
        }
    }

    int opr_swap = n - highest_count;
    int opr_copy = 0;
    int swapped = 0;

    while (swapped < opr_swap) {
        opr_copy++;
        swapped += highest_count;
        highest_count *= 2;
    }

    std::cout << opr_swap + opr_copy << '\n';
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int t;
    std::cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}

File: codeforces/python/B_Array_Cloning_Technique.py
# TLE again
# wtf python
# I will try this with c++ later


def solve():
    len_arr = int(input())
    arr = map(int, input().split())

    highest_common = 0
    num_count = {}
    for num in arr:
        num_count.setdefault(num, 0)
        num_count[num] += 1
        highest_common = max(highest_common, num_count[num])

    opr_swap = len_arr - highest_common
    opr_copy = 0
    swapped = 0

    while swapped < opr_swap:
        opr_copy += 1
        swapped += highest_common
        highest_common *= 2

    return opr_swap + opr_copy


def main():
    t = int(input())
    for _ in range(t):
        print(solve())


if __name__ == "__main__":
    main()
